src/snowflake/snowpark/modin/plugin/_internal/pivot_utils.py

```diff
@@ -1236,26 +1250,28 @@ def expand_pivot_result_with_pivot_table_margins_no_groupby_columns(
     pivot_snowflake_quoted_identifiers: list[str],
     values: list[str],
     margins_name: str,
+    original_aggfunc: AggFuncType,
 ) -> "SnowflakeQueryCompiler":  # type: ignore[name-defined] # noqa: F821
     names = pivot_qc.columns.names
     margins_frame = pivot_helper(
         original_modin_frame,
         pivot_aggr_groupings,
         not dropna,
-        not isinstance(aggfunc, list),
+        not is_list_like(aggfunc),
```

Comment:
Do we need to add the check for dict here? i.e. is_dict_like(aggfunc) or not is_list_like(aggfunc)?